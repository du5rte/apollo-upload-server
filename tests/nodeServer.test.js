import server from './nodeServer'
import upload from './upload'

const PORT = 1337
const serverURL = `http://localhost:${PORT}`

describe('nodeServer', () => {
  test('processRequest', async () => {
    expect.assertions(1)

    server.listen(PORT)

    try {
      const result = await upload(serverURL)

      expect(result).toMatchSnapshot();
    } catch (error) {
      console.error(error.message)
    } finally {
      server.close()
    }
  })
})
