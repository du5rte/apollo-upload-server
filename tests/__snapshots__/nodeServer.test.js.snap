// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodeServer processRequest 1`] = `
"{ query: 'mutation uploadAvatarMutation($id: ID! $avatar: Upload!) {uploadAvatar(id: $id avatar: $avatar) }',
  variables: 
   { id: '598645558d905ae18524bc55',
     avatar: 
      Stream {
        domain: null,
        _events: {},
        _eventsCount: 0,
        _maxListeners: undefined,
        readable: true,
        headers: [Object],
        name: 'avatar.gif',
        filename: 'avatar.gif',
        mime: 'image/gif',
        transferEncoding: 'binary',
        transferBuffer: '',
        size: 317866,
        type: 'image/gif' } },
  operationName: 'uploadAvatarMutation' }"
`;
